File: app/lib/app/models/teacher_model.dart



class Teacher {
  final String teacher_id;
  final String teacher_name;
  final String type;
  final String deptId;

  Teacher({required this.teacher_id, required this.teacher_name, required this.type, required this.deptId});


  factory Teacher.fromJson(Map<String, dynamic> json) {
    return Teacher(
      teacher_id: json['teacher_id'],
      teacher_name: json['teacher_name'],
      type: json['type'],
      deptId: json['dept_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'teacher_id': teacher_id,
      'teacher_name': teacher_name,
      'type': type,
      'dept_id': deptId,
    };
  }
}

File: backend/app/db/models/teacher_model.py
from typing import Optional
from pydantic import BaseModel
from uuid import uuid4
class TeacherCreate(BaseModel):
    teacher_id: Optional[str] = None
    teacher_name: str
    type: str
    dept_id: str
    
    def __init__(self, **data):
        if 'teacher_id' not in data or data['teacher_id'] is None:
            data['teacher_id'] = str(uuid4())
        super().__init__(**data)


# Request model for displaying a teacher
class DisplayTeacherRequest(BaseModel):
    teacher_id: str
    
class ReturnTeacherDetails(BaseModel):
    success : bool
    teachers : list[TeacherCreate] = []
    
class TeacherID(BaseModel):
    teacher_id: str
    
class TeacherEditRequest(BaseModel):
    teacher_id: Optional[str] = None
    teacher_name: Optional[str] = None
    type: Optional[str] = None
    
class UpdateTeacherRequest(BaseModel):
    previous_teacher_id: str
    details : TeacherEditRequest
